components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/RegularExpressionProtector.java

```diff
@@ -83,19 +104,93 @@ public boolean mediate(MessageContext messageContext) {
         if (messageProperty != null) {
             threatType = String.valueOf(messageProperty);
         }
-        checkRequestBody(messageContext);
-        checkRequestHeaders(messageContext);
-        checkRequestPath(messageContext);
+        if (isRequestBodyVulnerable(messageContext) || isRequestHeadersVulnerable(messageContext) ||
+                isRequestPathVulnerable(messageContext)) {
+            return true;
+        }
         return true;
     }
 
+    /**
+     * Using Regex Threat Protector mediator will be restricted to the tenants defined by the system property
+     * 'regexThreatProtectorEnabledTenants' as a list of comma separated values and super tenant. If this system
+     * property is not defined, then this restriction will not be applied at all. If invoked API is existing within a
+     * tenant, which was defined in this list, this method returns true. If this system property is not defined, this
+     * check won't be done and so will return true, hence all the tenants will be allowed to use this mediator
+     *
+     * @param messageContext contains the message properties of the relevant API request which was
+     *                       enabled the regexValidator message mediation in flow.
+     * @return true if the tenant is allowed to use this Mediator
+     */
+    private boolean isTenantAllowed(MessageContext messageContext) {
+        String allowedTenants = System.getProperty(APIMgtGatewayConstants.REGEX_THREAT_PROTECTOR_ENABLED_TENANTS);
+        if (allowedTenants == null) {
+            return true;
+        }
+        List<String> allowedTenantsList = Arrays.asList(allowedTenants.split(","));
+        String tenantDomain = MultitenantUtils.getTenantDomainFromRequestURL(RESTUtils.getFullRequestPath
+                (messageContext));
+        if (StringUtils.isEmpty(tenantDomain)) {
+            tenantDomain = org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+        if (!allowedTenantsList.contains(tenantDomain) &&
+                !tenantDomain.equals(org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
```

Comment:
Shall we swap the condition as follows to avoid NPE?

org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)

Reply:
Resolved